packages/nextjs/src/server/getAuth.ts

```diff
@@ -8,12 +8,14 @@ import {
   signedInAuthObject,
   signedOutAuthObject,
 } from '@clerk/backend';
+import type { SecretKeyOrApiKey } from '@clerk/types';
 
 import { API_KEY, API_URL, API_VERSION, SECRET_KEY } from './clerk';
 import type { RequestLike } from './types';
 import { getAuthKeyFromRequest, getCookie, getHeader, injectSSRStateIntoObject } from './utils';
 
-export const getAuth = (req: RequestLike): SignedInAuthObject | SignedOutAuthObject => {
+type GetAuthOpts = Partial<SecretKeyOrApiKey>;
```

Comment:
This is done so that, if we add more options there in the future, passing a key will still be optional.

Reply:
I think that in the future we should use a base type defined in the `@clerk/types` or `@clerk/backend` packages and omit or extend some options based on the limitations the SDK (eg cannot add `LoadResourcesOptions` on nextJS since middleware can be run in different machine).